bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -134,15 +159,51 @@ public String authority() {
   public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
       MethodDescriptor<ReqT, RespT> methodDescriptor, CallOptions callOptions) {
     Preconditions.checkState(!shutdown, "Cannot perform operations on a closed connection");
-    ClientCall<ReqT, RespT> delegate = getNextChannel().newCall(methodDescriptor, callOptions);
+    int currentRequestNum = requestCount.getAndIncrement();
+    ImmutableList<ChannelAndMetrics> channelsList = channels.get();
+    int index = Math.abs(currentRequestNum % channelsList.size());
```

Comment:
Why nuke the getNextChannel() logic here? NBD but personally I think it is a bit more readable to keep the modulus/round-robin logic in a separate function. 

Separately it makes me a *bit* nervous that there's nothing inherently ensuring that the channel we choose here is actually valid AFAICT? I guess the channelsList only ever grows, so it's not like we'll get an invalid index, but don't we want to ensure that the channel we select isn't already shutdown (and wouldn't doing that correctly require more synchronization)?